`tests/test_http_writer.py::test_write_large_payload_deflate_compression_data_in_eof[pyloop]` fails with Python 3.12.9

### Describe the bug

After upgrading to Python 3.12.9, the following test started failing:

```pytb
[gw0] [100%] FAILED tests/test_http_writer.py::test_write_large_payload_deflate_compression_data_in_eof[pyloop] 
```

I've reproduced both with aiohttp 3.11.12 and git @ e6ed541838d1f65085dc696cc8e587caa9e2d9fb.

### To Reproduce

1. Install Python 3.12.9.
2. `pytest -n1 tests/test_http_writer.py::test_write_large_payload_deflate_compression_data_in_eof`

### Expected behavior

Tests passing.

### Logs/tracebacks

```python-traceback
$ pytest -n1 tests/test_http_writer.py::test_write_large_payload_deflate_compression_data_in_eof 
========================================================= test session starts =========================================================
platform linux -- Python 3.12.9, pytest-8.1.1, pluggy-1.5.0 -- /tmp/aiohttp/.venv/bin/python
codspeed: 3.2.0 (disabled, mode: walltime, timer_resolution: 1.0ns)
cachedir: .pytest_cache
rootdir: /tmp/aiohttp
configfile: setup.cfg
plugins: cov-6.0.0, codspeed-3.2.0, mock-3.14.0, xdist-3.6.1
1 worker [1 item]      
scheduling tests via LoadScheduling

tests/test_http_writer.py::test_write_large_payload_deflate_compression_data_in_eof[pyloop] 
[gw0] [100%] FAILED tests/test_http_writer.py::test_write_large_payload_deflate_compression_data_in_eof[pyloop] 

============================================================== FAILURES ===============================================================
__________________________________ test_write_large_payload_deflate_compression_data_in_eof[pyloop] ___________________________________
[gw0] linux -- Python 3.12.9 /tmp/aiohttp/.venv/bin/python

protocol = <NonCallableMagicMock spec_set='BaseProtocol' id='139909514918992'>
transport = <NonCallableMagicMock name='mock.transport' spec_set='Transport' id='139909512338736'>
loop = <_UnixSelectorEventLoop running=False closed=False debug=False>

    async def test_write_large_payload_deflate_compression_data_in_eof(
        protocol: BaseProtocol,
        transport: asyncio.Transport,
        loop: asyncio.AbstractEventLoop,
    ) -> None:
        msg = http.StreamWriter(protocol, loop)
        msg.enable_compression("deflate")
    
        await msg.write(b"data" * 4096)
        assert transport.write.called  # type: ignore[attr-defined]
        chunks = [c[1][0] for c in list(transport.write.mock_calls)]  # type: ignore[attr-defined]
        transport.write.reset_mock()  # type: ignore[attr-defined]
    
        # This payload compresses to 20447 bytes
        payload = b"".join(
            [bytes((*range(0, i), *range(i, 0, -1))) for i in range(255) for _ in range(64)]
        )
        await msg.write_eof(payload)
        chunks.extend([c[1][0] for c in list(transport.write.mock_calls)])  # type: ignore[attr-defined]
    
        assert all(chunks)
        content = b"".join(chunks)
>       assert zlib.decompress(content) == (b"data" * 4096) + payload
E       zlib.error: Error -5 while decompressing data: incomplete or truncated stream

chunks     = [b'x\x9c']
content    = b'x\x9c'
loop       = <_UnixSelectorEventLoop running=False closed=False debug=False>
msg        = <aiohttp.http_writer.StreamWriter object at 0x7f3f38c896a0>
payload    = b'\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00\x01\x00...\x1f\x1e\x1d\x1c\x1b\x1a\x19\x18\x17\x16\x15\x14\x13\x12\x11\x10\x0f\x0e\r\x0c\x0b\n\t\x08\x07\x06\x05\x04\x03\x02\x01'
protocol   = <NonCallableMagicMock spec_set='BaseProtocol' id='139909514918992'>
transport  = <NonCallableMagicMock name='mock.transport' spec_set='Transport' id='139909512338736'>

tests/test_http_writer.py:148: error
```

### Python Version

```console
$ python --version
Python 3.12.9
```

### aiohttp Version

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 4.0.0a2.dev0
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: 
Author-email: 
License: Apache-2.0
Location: /tmp/aiohttp/.venv/lib/python3.12/site-packages
Editable project location: /tmp/aiohttp
Requires: aiohappyeyeballs, aiosignal, frozenlist, multidict, propcache, yarl
Required-by:
```

### multidict Version

```console
$ python -m pip show multidict
Name: multidict
Version: 6.1.0
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /tmp/aiohttp/.venv/lib/python3.12/site-packages
Requires: 
Required-by: aiohttp, yarl
```

### propcache Version

```console
$ python -m pip show propcache
Name: propcache
Version: 0.2.1
Summary: Accelerated property cache
Home-page: https://github.com/aio-libs/propcache
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /tmp/aiohttp/.venv/lib/python3.12/site-packages
Requires: 
Required-by: aiohttp, yarl
```

### yarl Version

```console
$ python -m pip show yarl
Name: yarl
Version: 1.18.3
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /tmp/aiohttp/.venv/lib/python3.12/site-packages
Requires: idna, multidict, propcache
Required-by: aiohttp
```

### OS

Gentoo Linux amd64

### Related component

Server, Client

### Additional context

_No response_

### Code of Conduct

- [x] I agree to follow the aio-libs Code of Conduct

--- a/CHANGES/10423.packaging.rst
+++ b/CHANGES/10423.packaging.rst
@@ -1,0 +1,1 @@
+Fixed test ``test_write_large_payload_deflate_compression_data_in_eof_writelines`` failing with Python 3.12.9+ or 3.13.2+ -- by :user:`bdraco`.

--- a/tests/test_http_writer.py
+++ b/tests/test_http_writer.py
@@ -20,6 +20,12 @@ def enable_writelines() -> Generator[None, None, None]:
 
 
 @pytest.fixture
+def disable_writelines() -> Generator[None, None, None]:
+    with mock.patch("aiohttp.http_writer.SKIP_WRITELINES", True):
+        yield
+
+
+@pytest.fixture
 def force_writelines_small_payloads() -> Generator[None, None, None]:
     with mock.patch("aiohttp.http_writer.MIN_PAYLOAD_FOR_WRITELINES", 1):
         yield
@@ -123,6 +129,7 @@ async def test_write_payload_length(
     assert b"da" == content.split(b"\r\n\r\n", 1)[-1]
 
 
+@pytest.mark.usefixtures("disable_writelines")
 async def test_write_large_payload_deflate_compression_data_in_eof(
     protocol: BaseProtocol,
     transport: asyncio.Transport,

